crates/trippy-tui/src/report/json.rs

```diff
@@ -9,14 +11,24 @@ pub fn report<R: Resolver>(
     info: &TraceInfo,
     report_cycles: usize,
     resolver: &R,
+    privacy_max_ttl: u8,
 ) -> anyhow::Result<()> {
     let start_timestamp = chrono::Utc::now();
     let trace = super::wait_for_round(&info.data, report_cycles)?;
     let end_timestamp = chrono::Utc::now();
     let hops: Vec<Hop> = trace
         .hops()
         .iter()
-        .map(|hop| Hop::from((hop, resolver)))
+        .map(|hop| {
+            let mut hop_report = Hop::from((hop, resolver));
+            if privacy_max_ttl > 0 && hop_report.ttl <= privacy_max_ttl {
+                hop_report.hosts = Hosts(vec![Host {
+                    ip: IpAddr::from([0, 0, 0, 0]),
+                    hostname: String::from("[hidden]"),
```

Comment:
nit: I'd suggest using `**hidden**` to match the TUI. For the `csv` I understand why you picked `[hidden]` as `*` is not safe to use in csv files.